### DISCUSSION TITLE
About the formula in the MIVVAI version

### DISCUSSION THREAD

**Discussion by cs123951** [voxelmorph/voxelmorph#403] [General]
Hello, 
   I have read your paper:
A. V. Dalca, G. Balakrishnan, J. Guttag, and M. R. Sabuncu, “Unsupervised Learning for Fast Probabilistic Diffeomorphic Registration,” in Medical Image Computing and Computer Assisted Intervention – MICCAI 2018, vol. 11070,  Cham: Springer International Publishing, 2018, pp. 729–738. doi: [10.1007/978-3-030-00928-1_82](https://doi.org/10.1007/978-3-030-00928-1_82). 

I am confused about the semicolon(;) in the formula(4).
<img width="591" alt="image" src="https://user-images.githubusercontent.com/13451757/157842963-393e1e86-336c-495d-8613-843e1d5284ea.png">

In this former, for example, y is the parameters of p(x|z;y). But in fact, y is variable, just like x. Given x and y, we will find p(x|z,y).  We do not need to change y.
see https://www.toolsou.com/en/article/220276803 
Only in this way, the second line in (4) can be derived to the third line in (4) using p(z|x,y)=p(x,y,z)/p(x,y).

Besides, formula (5) has also verified my assumption.
<img width="445" alt="image" src="https://user-images.githubusercontent.com/13451757/157845035-daf6bc33-fb00-4b30-8f63-67dd085ed2e4.png">
the z|x;y in the left should be changed to z|x,y as it is in the right. 

So I assume that some of the semicolons( ; ) should be changed to comma( , ). Isn't it?

**Comment by adalca** [comment]
We were using semi-colon to indicate that y is fixed in this formulation. 
p(z,x,y) should have a semi-colon p(z,x;y) undet that notation.
You could take y out of all these terms and have the same derivation, but adding the y there reminds us that there is a fixed image that we're doing everything with respect to.

**Reply by cs123951** [reply]
Thank you for your kind reply. This really solved my problem!

However, for paper, 
A. V. Dalca, G. Balakrishnan, J. Guttag, and M. R. Sabuncu, “Unsupervised Learning for Fast Probabilistic Diffeomorphic Registration,” in Medical Image Computing and Computer Assisted Intervention – MICCAI 2018, vol. 11070, Cham: Springer International Publishing, 2018, pp. 729–738. doi: [10.1007/978-3-030-00928-1_82](https://doi.org/10.1007/978-3-030-00928-1_82).

I am confused with another question.
I noticed that you have used VAE to solve the image registration problem. That is an inspired idea. 
However, during test time, since we sample \epsilon from the normal distribution with the reparameterization trick in VAE, this may bring some uncertainty, thus may cause different deformation fields. Is it possible? Did you ever consider this situation?  

Thank you again for your patient reading and look forward to your reply.

Sincerely,
Jiayi Lu

**Comment by adalca** [comment]
yes, we did, and in fact even play with the uncertainty maps that result from this, see the arxiv version figures at the end -- https://arxiv.org/pdf/1805.04605.pdf.

I would note that while the development is similar to how the VAE was derived, this is not a VAE since we don't auto-encode anything. It's just a variational approximation that uses a neural network

**Reply by cs123951** [reply]
Oh, I am sorry that I did not notice the uncertainty analysis. 
Your reply really solved my problem!
Thank you again for your patience~
